### PR TITLE
Add ServiceWorker externs to html5.js

### PR DESCRIPTION
**Pull request by nicks**

### CODE REVIEW

File: externs/html5.js [hunk 1]
```diff
@@ -3242,3 +3242,104 @@ function HTMLTemplateElement() {}
...
  */
 HTMLTemplateElement.prototype.content;
+
+
+/**
+ * @constructor
+ * @extends {Worker}
+ * @see http://www.w3.org/TR/service-workers/#service-worker-obj
+ */
+function ServiceWorker() {}
```
Comment (ChadKillingsworth) [inline externs/html5.js]: Should this be an interface? Would you ever call "new ServiceWorker()"?

Comment (nicks) [inline externs/html5.js]: i don't think it can be, because it extends Worker?

File: externs/html5.js [hunk 2]
```diff
@@ -3242,3 +3242,104 @@ function HTMLTemplateElement() {}
...
+ServiceWorker.prototype.onstatechange;
+
+
+
+/**
+ * @constructor
+ * @implements {EventTarget}
+ * @see http://www.w3.org/TR/service-workers/#service-worker-registration-obj
+ */
+function ServiceWorkerRegistration() {}
```
Comment (ChadKillingsworth) [inline externs/html5.js]: Is there a reason to omit the `installing`, `waiting` and `active` properties?

File: externs/html5.js [hunk 3]
```diff
@@ -3242,3 +3242,104 @@ function HTMLTemplateElement() {}
...
+ServiceWorkerContainer.prototype.addEventListener;
+
+/** @override */
+ServiceWorkerContainer.prototype.removeEventListener;
+
+/** @override */
+ServiceWorkerContainer.prototype.dispatchEvent;
+
+/** @type {!ServiceWorker} */
+ServiceWorkerContainer.prototype.controller;
```
Comment (ChadKillingsworth) [inline externs/html5.js]: I think this is supposed to be nullable.

File: externs/html5.js [hunk 4]
```diff
@@ -3242,3 +3242,104 @@ function HTMLTemplateElement() {}
...
+
+/**
+ * @param {string} scriptURL
+ * @param {{scope: string}=} options
+ * @return {!Promise<ServiceWorkerRegistration>}
+ */
+ServiceWorkerContainer.prototype.register = function (scriptURL, options) {};
+
+/**
+ * @param {string} clientURL
```
Comment (ChadKillingsworth) [inline externs/html5.js]: This is an optional param

File: externs/html5.js (hunk 1 above)
Comment (ChadKillingsworth) [inline externs/html5.js]: You are correct. I thought that might be allowed in externs, but no:

> JSC_CONFLICTING_EXTENDED_TYPE: ServiceWorker cannot extend this type; interfaces can only extend interfaces at line 11 character 0
function ServiceWorker() {}

**Comment by MatrixFrog** [comment]
We already have some ServiceWorker externs in our internal repo. Didn't realize that we forgot to put them on the open-source-able list.

**Comment by nicks** [comment]
ah, ok :+1:

**Pull request closed without merging**